addons/crm/views/crm_lead_views.xml

```diff
@@ -574,7 +574,12 @@
                                     <field name="recurring_plan" groups="crm.group_use_recurring_revenues"/>
                                 </t>
                             </div>
-                            <field name="partner_id" class="text-truncate" />
+                            <div class="d-flex" invisible="not partner_id">
+                                <field name="partner_id" widget="many2one_avatar" class="text-truncate" readonly="1" options="{'no_create': 1, 'no_open': 1}"/>
+                                <field name="partner_id" class="ms-2 text-truncate"/>
+                            </div>
+                            <field name="contact_name" invisible="partner_id"/>
+                            <field name="partner_name" invisible="partner_id or contact_name"/>
```

Comment:
Note it is the company name, so maybe it should be displayed if we don't have partner_id (as it is complementary to contact_name, and will be used at convert time to create a company). To check with LNA probably :)